polar-core/src/filter.rs

```diff
@@ -357,8 +354,24 @@ impl FilterInfo {
         self.add_condition(left, Comparison::In, right)
     }
 
+    /// Validate FilterInfo before constructing a Filter
+    fn validate(self, root: &str) -> FilterResult<Self> {
+        let mut set = singleton(root);
+        for Relation(_, _, dst) in self.relations.iter() {
+            if set.contains(dst as &str) {
+                return invalid_state_error(format!(
```

Comment:
Should this be an unsupported error? I thought invalid state errors are for when we have an internal bug that ends up breaking one of our invariants? This could be user caused though.

Reply:
Unsupported errors need a `term`, which we don't have here.